std/root.ln

```diff
@@ -804,7 +1064,7 @@ export fn prune(n: Node<any>): Tree<any> {
       const childrenIds = childrenRes.getR();
       nodeStack = nodeStack.concat(childrenIds);
     }
-    const delIdx = nodeId - length(rmdIds);
+    const delIdx = nodeId - length(rmdIds) || 0;
     delete(n.tree.vals, delIdx);
     delete(n.tree.parents, delIdx);
     delete(n.tree.children, delIdx);
```

Comment:
But array deletion can fail if the index is out-of-bounds, so it feels like this should accept a Result-wrapped index since it can just fail early if the input index is an error.